src/module/system/damage/roll.ts

```diff
@@ -492,11 +492,24 @@ class DamageInstance extends AbstractDamageRoll {
     }
 
     /** Get the total of this instance without any doubling or tripling from a critical hit */
-    get critImmuneTotal(): number | undefined {
+    get critImmuneTotal(): this["total"] {
         if (!this._evaluated) return undefined;
 
         const { head } = this;
-        return head instanceof ArithmeticExpression || head instanceof Grouping ? head.critImmuneTotal : this.total;
+
+        // Get the total with all damage-doubling removed
+        const undoubledTotal =
+            head instanceof ArithmeticExpression || head instanceof Grouping ? head.critImmuneTotal : this.total;
+
+        if (game.settings.get("pf2e", "critRule") === "doubledamage") {
+            return undoubledTotal;
```

Comment:
it seems you check for the crit rule twice, once when building the total, and again here in the damage roll. Since it being double dice seems to be baked into the formula itself, perhaps this is something that should be set in roll options.